Feat: Add button to filter activities by type.

1andreea26 opened the pull request:
## Summary

This PR introduces a button in the ActivitiesScreen, which when pressed, opens a dialog allowing the user to select one or more Activity types to filter the Activities displayed. When pressed confirm, dialog closes and only the filtered trips are displayed. The user is also able to de-select one or more filters previously selected.
Moreover, documentation has been updated for the ActivitiesScreen Composable.

## Changes

- **Screens/UI:** Filter button appears now in ActiviesScreen, a Dialog Composable has been implemented in ActivityUtils and is displayed in ActivitiesScreen by pressing the button.

## Checklist

Ensure all / most of these are checked before the pull request is submitted.
- [X] This PR resolves #229.
- [X] Tests have been added for new functionality.
- [X] Screenshots or UI changes have been attached (if applicable).
- [X] Documentation has been updated (if applicable).

##  Testing

- **Manual Testing:**
    - [X] Tested on emulator / physical device.
- **Unit Tests**
    - [X] Added units tests for this

##  Related Issues/Tickets

Closes #229.

## Screenshots (if applicable)


https://github.com/user-attachments/assets/b5e13ddb-d829-4700-884b-4bd028406f87



## 💡 Additional Context

The filter button has been added in a Top bar. In the same Top bar should appear the Search for Activities. For the moment, the Title of the Top bar is "Activities bar", but it should be replaced by the Search.

ioanajianu commented on app/src/main/java/com/android/voyageur/ui/trip/activities/ActivityUtils.kt:
```diff
@@ -63,3 +74,52 @@ fun DeleteActivityAlertDialog(
+      },
+      text = {
+        Column(
+            modifier = Modifier.fillMaxWidth(), verticalArrangement = Arrangement.spacedBy(10.dp)) {
```
To me the spacing between rows is quite big, wdyt of making it a bit smaller?

1andreea26 replied:
You're right :) It looks better with smaller spacing! I'll make the change.

ioanajianu commented on app/src/main/java/com/android/voyageur/ui/trip/activities/ActivityUtils.kt:
```diff
@@ -63,3 +74,52 @@ fun DeleteActivityAlertDialog(
+                    verticalAlignment = Alignment.CenterVertically,
+                    horizontalArrangement = Arrangement.SpaceBetween) {
+                      Text(
+                          text = type.name,
```
Wdyt about changing this to capital letter and then small letters instead of all capital letters? (Restaurant instead of RESTAURANT)
I believe it would look a bit cleaner, but I leave this up to you :)

1andreea26 replied:
I think you're right as well here! Feels a bit friendlier to have lowercase instead of capital letters. I'll modify this as well.

ioanajianu commented on app/src/androidTest/java/com/android/voyageur/ui/trip/activities/ActivitiesScreenTest.kt:
```diff
@@ -176,4 +176,65 @@ class ActivitiesScreenTest {
+    }
+    // Click the filter button
+    composeTestRule.onNodeWithTag("filterButton").performClick()
+    composeTestRule.onNodeWithTag("filterActivityAlertDialog").assertIsDisplayed()
```
here can you perform a click somewhere else and check if the dialog stopped being displayed?

ioanajianu commented on app/src/androidTest/java/com/android/voyageur/ui/trip/activities/ActivitiesScreenTest.kt:
```diff
@@ -176,4 +176,65 @@ class ActivitiesScreenTest {
+
+    composeTestRule.onNodeWithTag("filterButton").performClick()
+    // Select the "WALK" filter
+    composeTestRule.onNodeWithTag("typeCheckBox_WALK").performClick()
```
You could also assert that the checkbox is checked

ioanajianu reviewed (changes requested):
Thank you Andreea for implementing this, the new feature works well and looks nice :)
The code is well written, I liked the approach you used when displaying only the filtered activities. The documentation and tests you added look good!
I left only some minor comments, otherwise great work!!

ioanajianu reviewed (approved):
Thanks Andreea for making the changes I suggested and implementing this feature! Looks great!

1andreea26 commented:
> Thank you Andreea for implementing this, the new feature works well and looks nice :) The code is well written, I liked the approach you used when displaying only the filtered activities. The documentation and tests you added look good! I left only some minor comments, otherwise great work!!

Thanks for the review Ioana! I implemented the changes you suggested in your comments and also added the suggested tests.
